relentless/optimize/objective.py

```diff
@@ -165,3 +181,150 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    r"""Relative entropy.
+
+    The relative entropy quantifies how different two probability distributions are.
+    This can be applied to the probability distributions of a target ensemble
+    and the ensemble of a simulated molecular system, which is parametrized
+    by some pair potential functions.
+
+    The value of the relative entropy is not determined. The gradient of the relative
+    entropy is equivalent to the gradient of the likelihood of the design variables
+    :math:`\mathbf{x}` given the target ensemble :math:`\mathbf{T}`. Given
+    the rdfs of the simulation and target ensembles, :math:`g(r|\mathbf{x})` and :math:`g_{tgt}(r)`,
+    respectively, the total pair potential energy, the gradient is:
+
+    .. math::
+
+        \nabla_\mathbf{x} L\left(\mathbf{x}|\mathbf{T}\right) = 2\pi N\rho\int{dr r^2 \left(g(r|\mathbf{x})-g_{tgt}(r)\right)\beta u(r|\mathbf{x})}
+
+    where :math:`\beta` is the thermodynamic beta, :math:`N` is the particular number,
+    and :math:`\rho=N/V` is the number density, all for the target ensemble.
+
+    Parameters
+    ----------
+    target : :class:`~relentless.ensemble.Ensemble`
+        The target ensemble (must have specified N and V).
+    simulation :class:`~relentless.simulate.Simulation`
+        The simulation engine to use, with specified simulation operations.
+    potentials :class:`~relentless.simulate.Potentials`
+        The pair potentials to use in the simulations.
+    thermo : :class:`~relentless.simulate.SimulationOperation`
+        The thermodynamic analyzer for the simulation ensemble and rdf (usually
+        :meth:`AddEnsembleAnalyzer()` for the specified simulation type.
+    dr : float
+        The radial step size.
+    communicator : :class:`~relentless.mpi.Communicator`
+        The communicator used for saving outputs to file ##?## (defaults to
+        ``None``).
+
+    """
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        """Evaluate the value and gradient of the relative entropy function.
+
+
```

Comment:
You should add a comment here about what is going to be written and how the files will be named.

I would also add a comment making it clear that this will require running a simulation, so it can be expensive to compute.